gui/views/pattern_map/PatternMap.gd

```diff
@@ -753,6 +771,14 @@ func _set_pattern_at_cursor(pattern_idx: int) -> void:
 				current_arrangement.set_pattern(bar_index, cell.y, value)
 	)
 	
+func _set_pattern_at_position(at_position: Vector2, pattern_idx: int) -> void:
```

Comment:
If I didn't miss anything, this method is only used in `_set_pattern_at_cursor`, which is a one-line method. There is no point introducing this abstraction at this time, you should put all this directly in `_set_pattern_at_cursor`.